build.gradle

```diff
@@ -219,6 +219,58 @@ project(':iceberg-hive') {
   }
 }
 
+project(':iceberg-mr') {
+  dependencies {
+    compile project(':iceberg-api')
+    compile project(':iceberg-core')
+    compile project(':iceberg-orc')
+    compile project(':iceberg-parquet')
+    compile project(':iceberg-data')
+
+    compileOnly("org.apache.hadoop:hadoop-client") {
+      exclude group: 'org.apache.avro', module: 'avro'
+    }
+
+    testCompile project(':iceberg-hive')
+    testCompile project(path: ':iceberg-data', configuration: 'testArtifacts')
+    testCompile project(path: ':iceberg-hive', configuration: 'testArtifacts')
+    testCompile project(path: ':iceberg-api', configuration: 'testArtifacts')
+    testCompile project(path: ':iceberg-core', configuration: 'testArtifacts')
+
+    // By default, hive-exec is a fat/uber jar and it exports a guava library
+    // that's really old. We use the core classifier to be able to override our guava
+    // version. Luckily, hive-exec seems to work okay so far with this version of guava
+    // See: https://github.com/apache/hive/blob/master/ql/pom.xml#L911 for more context.
+    testCompile("org.apache.hive:hive-exec::core") {
```

Comment:
Are Hive dependencies still needed for `testCompile`?

To test this with HMS, we could make it a test dependency of `iceberg-hive` and test out the catalog function.

Reply:
I'l also worried that iceberg-hive in test scope can create a circular dep. Maybe we can just test with `HadoopCatalog` ?

Reply:
Why would it create a circular dependency? This doesn't need to depend on iceberg-hive anymore, right?

Reply:
We'd need to depend on iceberg-hive in test scope so that we can use HiveCatalog

Reply:
What about putting the test case for HiveCatalog in iceberg-hive using the catalog function? Then we get a test that uses the catalog plugin system and we don't have to include Hive here.

Reply:
I've remove the iceberg-hive and related deps from iceberg-mr module and the plugin system is being tested using `HadoopCatalog`.  I can still add the test for HiveCatalog in the iceberg-hive module. What do you think?